File: lib/rest/yampi/mappers/yampi_banner_mapper.dart
import 'package:sertton/core/marketing/dtos/banner_dto.dart';
import 'package:sertton/rest/types/json.dart';

class YampiBannerMapper {
  static BannerDto toDto(Json json) {
    return BannerDto(id: json['id'].toString(), imageUrl: json['image_url']);
  }

  static List<BannerDto> toDtoList(Json json) {
    final data = (json['data'] as List).cast<Json>();
    return data.map(toDto).toList();
  }
}

File: src/rest/yampi/mappers/yampi-banner-mapper.ts
import type { Banner } from "@/core/marketing/entities"
import type { YampiBanner } from "@/rest/yampi/types"

export const YampiBannerMapper = () => ({
  toDomain(input: YampiBanner): Banner {
    return {
      id: String(input.id),
      imageUrl: input.image_url.startsWith("//") ? `https:${input.image_url}` : input.image_url,
    }
  },
})
